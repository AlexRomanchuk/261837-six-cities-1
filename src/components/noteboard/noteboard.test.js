import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Noteboard from "../noteboard/noteboard.jsx";

const mockPlaces = [
  {
    id: 999,
    isPremium: true,
    picture: `test-image1.jpg`,
    price: 111,
    rating: 22,
    title: `Test title 1`,
    description: `text`,
    bedrooms: 2,
    type: `apartment`,
    goods: [`one`, `two`, `tree`],
    images: [`url1`, `url2`, `url3`],
    previewImage: ``,
    maxAdults: 2,
    isFavorite: true,
    city: {
      name: `City name 1`,
      location: {
        latitude: 1,
        longitude: 2,
        zoom: 13
      }
    },
    location: {
      latitude: 1,
      longitude: 2,
      zoom: 13
    },
    host: {
      id: 1,
      name: `name`,
      isPro: true,
      avatarUrl: `url`
    }
  },
  {
    id: 222,
    isPremium: false,
    picture: `test-image2.jpg`,
    price: 222,
    rating: 100,
    title: `Test title 2`,
    description: `text`,
    goods: [`one`, `two`, `tree`],
    images: [`url1`, `url2`, `url3`],
    bedrooms: 3,
    maxAdults: 1,
    type: `house`,
    previewImage: ``,
    isFavorite: true,
    coordinates: [52.3909553943508, 4.85309666406198],
    city: {
      name: `City name 2`,
      location: {
        latitude: 1,
        longitude: 2,
        zoom: 13
      }
    },
    location: {
      latitude: 1,
      longitude: 2,
      zoom: 13
    },
    host: {
      id: 1,
      name: `name`,
      isPro: true,
      avatarUrl: `url`
    }
  },
];

const sortParameters = [
  {
    name: `Popular`,
    action: null
  },
  {
    name: `Price: low to high`,
    action: (place1, place2) => place1.price - place2.price
  },
  {
    name: `Price: high to low`,
    action: (place1, place2) => place2.price - place1.price
  },
  {
    name: `Top rated first`,
    action: (place1, place2) => place2.rating - place1.rating
  }
];

const onClick = () => {
  return true;
};

it(`correct renders noteboard page`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(<Noteboard
    city={`Amsterdam`}
    cityCoords={{location: {
      coords: [52.3809553943508, 4.939309666406198],
    }}}
    isSortOpen={false}
    activeParameter={sortParameters[0]}
    parameters={sortParameters}
    places={mockPlaces}
    activePlace={mockPlaces[1]}
    onClick={onClick}
    onSortClick={onClick}
    onOpenSortClick={onClick}
    onSelect={() => {}}
  />);
  expect(tree).toMatchSnapshot();
});
