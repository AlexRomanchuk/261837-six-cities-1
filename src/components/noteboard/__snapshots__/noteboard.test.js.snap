// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct renders noteboard page 1`] = `
<div
  className="cities__places-wrapper"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
        2
         places to stay in 
        Amsterdam
      </b>
      <div
        action="#"
        className="places__sorting"
        method="get"
      >
        <span
          className="places__sorting-caption"
        >
          Sort byÂ 
        </span>
        <span
          className="places__sorting-type"
          onClick={[Function]}
          tabIndex="0"
        >
          Popular
          <svg
            className="places__sorting-arrow"
            height="4"
            width="7"
          >
            <use
              xlinkHref="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          className="places__options places__options--custom "
          onClick={[Function]}
        >
          <li
            className="places__option places__option--active"
            onClick={[Function]}
            tabIndex="0"
          >
            Popular
          </li>
          <li
            className="places__option "
            onClick={[Function]}
            tabIndex="0"
          >
            Price: low to high
          </li>
          <li
            className="places__option "
            onClick={[Function]}
            tabIndex="0"
          >
            Price: high to low
          </li>
          <li
            className="places__option "
            onClick={[Function]}
            tabIndex="0"
          >
            Top rated first
          </li>
        </ul>
      </div>
      <div
        className="cities__places-list places__list tabs__content"
      >
        <Memo(ConnectFunction)
          onClick={[Function]}
          place={
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 1,
                  "longitude": 2,
                  "zoom": 13,
                },
                "name": "City name 1",
              },
              "description": "text",
              "goods": Array [
                "one",
                "two",
                "tree",
              ],
              "host": Object {
                "avatarUrl": "url",
                "id": 1,
                "isPro": true,
                "name": "name",
              },
              "id": 999,
              "images": Array [
                "url1",
                "url2",
                "url3",
              ],
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 1,
                "longitude": 2,
                "zoom": 13,
              },
              "maxAdults": 2,
              "picture": "test-image1.jpg",
              "previewImage": "",
              "price": 111,
              "rating": 22,
              "title": "Test title 1",
              "type": "apartment",
            }
          }
        />
        <Memo(ConnectFunction)
          onClick={[Function]}
          place={
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 1,
                  "longitude": 2,
                  "zoom": 13,
                },
                "name": "City name 2",
              },
              "coordinates": Array [
                52.3909553943508,
                4.85309666406198,
              ],
              "description": "text",
              "goods": Array [
                "one",
                "two",
                "tree",
              ],
              "host": Object {
                "avatarUrl": "url",
                "id": 1,
                "isPro": true,
                "name": "name",
              },
              "id": 222,
              "images": Array [
                "url1",
                "url2",
                "url3",
              ],
              "isFavorite": true,
              "isPremium": false,
              "location": Object {
                "latitude": 1,
                "longitude": 2,
                "zoom": 13,
              },
              "maxAdults": 1,
              "picture": "test-image2.jpg",
              "previewImage": "",
              "price": 222,
              "rating": 100,
              "title": "Test title 2",
              "type": "house",
            }
          }
        />
      </div>
    </section>
    <div
      className="cities__right-section"
    >
      <section
        className="cities__map map"
      >
        <Map
          activePlace={
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 1,
                  "longitude": 2,
                  "zoom": 13,
                },
                "name": "City name 2",
              },
              "coordinates": Array [
                52.3909553943508,
                4.85309666406198,
              ],
              "description": "text",
              "goods": Array [
                "one",
                "two",
                "tree",
              ],
              "host": Object {
                "avatarUrl": "url",
                "id": 1,
                "isPro": true,
                "name": "name",
              },
              "id": 222,
              "images": Array [
                "url1",
                "url2",
                "url3",
              ],
              "isFavorite": true,
              "isPremium": false,
              "location": Object {
                "latitude": 1,
                "longitude": 2,
                "zoom": 13,
              },
              "maxAdults": 1,
              "picture": "test-image2.jpg",
              "previewImage": "",
              "price": 222,
              "rating": 100,
              "title": "Test title 2",
              "type": "house",
            }
          }
          city={
            Object {
              "location": Object {
                "coords": Array [
                  52.3809553943508,
                  4.939309666406198,
                ],
              },
            }
          }
          places={
            Array [
              Object {
                "bedrooms": 2,
                "city": Object {
                  "location": Object {
                    "latitude": 1,
                    "longitude": 2,
                    "zoom": 13,
                  },
                  "name": "City name 1",
                },
                "description": "text",
                "goods": Array [
                  "one",
                  "two",
                  "tree",
                ],
                "host": Object {
                  "avatarUrl": "url",
                  "id": 1,
                  "isPro": true,
                  "name": "name",
                },
                "id": 999,
                "images": Array [
                  "url1",
                  "url2",
                  "url3",
                ],
                "isFavorite": true,
                "isPremium": true,
                "location": Object {
                  "latitude": 1,
                  "longitude": 2,
                  "zoom": 13,
                },
                "maxAdults": 2,
                "picture": "test-image1.jpg",
                "previewImage": "",
                "price": 111,
                "rating": 22,
                "title": "Test title 1",
                "type": "apartment",
              },
              Object {
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 1,
                    "longitude": 2,
                    "zoom": 13,
                  },
                  "name": "City name 2",
                },
                "coordinates": Array [
                  52.3909553943508,
                  4.85309666406198,
                ],
                "description": "text",
                "goods": Array [
                  "one",
                  "two",
                  "tree",
                ],
                "host": Object {
                  "avatarUrl": "url",
                  "id": 1,
                  "isPro": true,
                  "name": "name",
                },
                "id": 222,
                "images": Array [
                  "url1",
                  "url2",
                  "url3",
                ],
                "isFavorite": true,
                "isPremium": false,
                "location": Object {
                  "latitude": 1,
                  "longitude": 2,
                  "zoom": 13,
                },
                "maxAdults": 1,
                "picture": "test-image2.jpg",
                "previewImage": "",
                "price": 222,
                "rating": 100,
                "title": "Test title 2",
                "type": "house",
              },
            ]
          }
        />
      </section>
    </div>
  </div>
</div>
`;
